### DISCUSSION TITLE
Can PyMuPDF add audio/mp3 file into the pdf so that I can play the mp3 inside pdf offline?

### DISCUSSION THREAD

**Discussion by bk111** [pymupdf/PyMuPDF#2519] [Ideas]
Can PyMuPDF add audio/mp3 file into the pdf so that I can play the mp3 inside pdf offline?

**Comment by JorjMcKie** [comment]
No sorry, adding this annotation type is currently not supported - only the extraction of audio streams works.

**Comment by JorjMcKie** [comment]
We are still investigating how to best support this.
However, we suggest to use the following approach which will deliver a solution that does work with most / many PDF readers. The basic idea is to use FileAttachmant annotations instead of audio attachments.
```python
import fitz
from pathlib import Path
filename = "some-folder/ping1.mp3"
stream = Path(filename).read_bytes()  # get audio file content
basename = Pathfilename).name
page.add_file_annot(point, stream, filename=basename, icon="Speaker")
```
This will add an annotation looking like this:
![grafik](https://github.com/pymupdf/PyMuPDF/assets/8290722/619f6c0a-d9a9-4f3f-a3c5-3b1bce41c06f)

Double clicking on it with readers like Adobe, Foxitand many more will offer you to "open" the file. This will them cause your standard media player to take over and play the file.

**Comment by JorjMcKie** [comment]
I am going to transfer this issue to a "Discussions" items, category "Ideas", so it won't be forgotten.
